hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/output/FileOutputCommitter.java

```diff
@@ -163,7 +168,11 @@ public FileOutputCommitter(Path outputPath,
       this.outputPath = fs.makeQualified(outputPath);
     }
   }
-  
+
+  private static void setJobPendingDirName(JobContext context) {
+    JOB_PENDING_DIR_NAME = "_temporary_" + context.getJobID();
```

Comment:
there's an assumption here that the job ID is always unique. this doesn''t hold for all spark versions. SPARK-33402